internal/namespaces/rdb/v1/custom_instance_connect.go

```diff
@@ -0,0 +1,202 @@
+	if instance == nil {
+		return Unknown, fmt.Errorf("instance engine is nil")
+	}
+	if strings.HasPrefix(instance.Engine, string(PostgreSQL)) {
+		return PostgreSQL, nil
+	}
+	if strings.HasPrefix(instance.Engine, string(MySQL)) {
+		return MySQL, nil
+	}
+	return Unknown, fmt.Errorf("unknown engine: %s", instance.Engine)
+}
+
+func createCommandLineArgs(instance *rdb.Instance, family engineFamily, args *instanceConnectArgs) ([]string, error) {
+	database := "rdb"
+
+	switch family {
+	case PostgreSQL:
+		tool := "psql"
+		if args.Database != nil {
+			database = *args.Database
+		}
+		if args.Tool != nil {
+			tool = *args.Tool
+		}
+
+		// psql -h 51.159.25.206 --port 13917 -d rdb -U username
+		return []string{
+			tool,
+			"--host", instance.Endpoint.IP.String(),
+			"--port", fmt.Sprintf("%d", instance.Endpoint.Port),
+			"--username", args.Username,
+			"--dbname", database,
+		}, nil
+	case MySQL:
+		tool := "mysql"
+		if args.Database != nil {
+			database = *args.Database
+		}
+		if args.Tool != nil {
+			tool = *args.Tool
+		}
+
+		// mysql -h 195.154.69.163 --port 12210 -p -u username
+		return []string{
+			tool,
+			"--host", instance.Endpoint.IP.String(),
+			"--port", fmt.Sprintf("%d", instance.Endpoint.Port),
+			"--database", database,
+			"--user", args.Username,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("unrecognize database engine: %s", instance.Engine)
+}
+
+func instanceConnectCommand() *core.Command {
+	return &core.Command{
+		Namespace: "rdb",
+		Resource:  "instance",
+		Verb:      "connect",
+		Short:     "Connect to an instance using local database cli",
+		Long:      "Connect to an instance using local database cli such as psql or mysql.",
+		ArgsType:  reflect.TypeOf(instanceConnectArgs{}),
+		ArgSpecs: core.ArgSpecs{
+			{
+				Name:       "instance-id",
+				Short:      `UUID of the instance`,
+				Required:   true,
+				Positional: true,
+			},
+			{
+				Name:     "username",
+				Short:    "Name of the user to connect with to the database",
+				Required: true,
+			},
+			{
+				Name:  "database",
+				Short: "Name of the database",
+			},
+			{
+				Name:  "tool",
+				Short: "Command line tool to use, default to psql/mysql",
+			},
+			core.RegionArgSpec(scw.RegionFrPar, scw.RegionNlAms),
+		},
+		Run: func(ctx context.Context, args interface{}) (interface{}, error) {
+			request := args.(*instanceConnectArgs)
+
+			client := core.ExtractClient(ctx)
+			api := rdb.NewAPI(client)
+			instance, err := api.GetInstance(&rdb.GetInstanceRequest{
+				Region:     request.Region,
+				InstanceID: request.InstanceID,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			engineFamily, err := detectEngineFamily(instance)
+			if err != nil {
+				return nil, err
+			}
+
+			cmd, err := createCommandLineArgs(instance, engineFamily, request)
+			if err != nil {
+				return nil, err
+			}
+
+			if !passwordFileExist(ctx, engineFamily) {
+				fmt.Println(passwordFileHint(engineFamily))
```

Comment:
- Don't print on stdout
- Use interactive.Println or logger for this type of usage